### ISSUE TITLE
Failures on "Remove spawning of DDS from DAP and let the spawned `dart run`/`dart test` do it"

### ISSUE THREAD

**Issue by derekxu16** [dart-lang/sdk#56965]
There are new test failures on [Remove spawning of DDS from DAP and let the spawned `dart run`/`dart test` do it](https://github.com/dart-lang/sdk/commit/97ab888201f4a7b36898bf418e79250449bc7e20).

The tests
```
pkg/dds/test/dap/integration/debug_breakpoints_test Timeout (expected Pass)
pkg/dds/test/dap/integration/debug_eval_test RuntimeError (expected Pass)
pkg/dds/test/dap/integration/debug_exceptions_test RuntimeError (expected Pass)
pkg/dds/test/dap/integration/debug_services_test RuntimeError (expected Pass)
pkg/dds/test/dap/integration/debug_stack_test RuntimeError (expected Pass)
pkg/dds/test/dap/integration/debug_variables_test RuntimeError (expected Pass)
```
are failing on configurations
```
unittest-asserts-release-win-arm64
```

cc @DanTup and @bkonyi

**Comment by DanTup** [comment]
All the tests that expect to pause on breakpoints seem to be pausing on isolate exits:

```
  Expected: 'breakpoint'
    Actual: 'exit'
```

Taking a look...

**Comment by DanTup** [comment]
@derekxu16 @bkonyi I'm failing to reproduce this on my Windows PC (although it's not ARM). Is it possible to trigger Windows ARM trybots on https://dart-review.googlesource.com/c/sdk/+/391965 to get a complete trace of the protocol traffic?

**Comment by DanTup** [comment]
Looking at the logs, this looks like a drive letter casing mismatch:

```
file:///c:/opt/s/w/ir/x/t/dart-sdk-dap-test8c56f93c/appd3adfb9c/test_file.dart
file:///C:/opt/s/w/ir/x/t/dart-sdk-dap-test8c56f93c/appd3adfb9c/test_file.dart
```

When trying to add the breakpoint, we have an uppercase C, but in the scripts list, there's a lowercase C. I can't see anything in the code removed in https://github.com/dart-lang/sdk/commit/97ab888201f4a7b36898bf418e79250449bc7e20 that would affect the drive letter. @bkonyi is there anything in how the VM spawns DDS that might affect this? I think I've seen lowercase drive letters in the VM script when the SDK was spawned using a lowercase drive letter, but there's nothing different in how we're spawning it here (we do force the drive letter uppercase after reading `Platform.resolvedExecutable` when the adapter spawns `dart run`).

**Comment by DanTup** [comment]
I was able to repro this locally by setting my Temp folder env var to a path with a lowercase drive letter. It repros with and without DDS, but it does not repro without the CL above.

I captured the protocol traffic and the first difference in the logs (that isn't just random tokens/port numbers/etc.) is the response to `getVM` where the working one has a reasonable path to the target script in the IsolateGroup name, and the other has a relative path that has likely been computed between paths with different drive letter casings:

![image](https://github.com/user-attachments/assets/d6386cb1-6f8b-4b28-9c0d-cd29db3a31ce)

So my guess is that _somewhere_ there is a `path.relative` call that may be operating on paths with different casings with/without the change above. I'll see if I can narrow it down.

**Comment by DanTup** [comment]
@bkonyi after much head scratching, I tracked this down to occurring only when the current working directory had a lowercase drive letter. It's not clear to me how the CL noted above changed the behaviour, but if the working directory has a lowercase drive letter, before this change the VM would handle breakpoints with uppercase drive letters, but after the change above, it would not.

I'm fixing this by having DAP normalize `Directory.current` to have an uppercase drive letter always (even though there's not strictly any requirement that the working directory is related to the package being run), but FYI in case it rings any bells.

**Comment by DanTup** [comment]
I believe https://dart-review.googlesource.com/c/sdk/+/392442 will resolve this.

**Comment by DanTup** [comment]
@bkonyi @derekxu16 seems like the fix above didn't work (it's possible the issue I repro'd locally that this fixes is not exactly the same as what's happening on the bots).

To help debug further, I've prepared two CLs that run just a single test in verbose mode again, with the same hard-coded paths for the test app that I'd like to run the Windows trybots on and compare the logs:

- https://dart-review.googlesource.com/c/sdk/+/392662
- https://dart-review.googlesource.com/c/sdk/+/393300

One of them includes backing out the original CL (that removed DDS from being spawned in-process) and the other does not. Both run the same test with the same path in verbose mode.

If we can trigger the Windows trybot for this (I added a `Cq-Include-Trybots` header I think will do that, but I've not used it before so I'm not certain) I'll review the logs for any differences and we'll go from there. I still feel like this could be related to the casing different in drive letters for the cwd, but my CL to force that didn't work, but I can try applying it to one of these CLs again after I have these results to compare.
